### PR TITLE
ControlPort: Start RPC on blocks that only have message connections

### PR DESCRIPTION
**Pull request by mrjacobagilbert**
This PR addresses https://github.com/gnuradio/gnuradio/issues/1110 by including logic to call setup_rpc() on blocks that only have message port connections.

### CODE REVIEW

**Comment by mrjacobagilbert** [comment]
For the record, it looks like this can be addressed in the for loop here https://github.com/gnuradio/gnuradio/blob/master/gnuradio-runtime/lib/hier_block2_detail.cc#L886 and have the rest of the code relating to starting RPC removed. I just can't seem to sort out all the iterators and sptrs to figure out exactly what the right calls are.

**Comment by jmcorgan** [comment]
That code is indeed pretty ugly; I think the guy who wrote it was high.

**Comment by bastibl** [comment]
I don't have a test case, but couldn't we add something like 
```
if(ctrlport_on) {
  if(!(*b)->is_rpc_set()) {
    (*b)->setup_rpc();
    (*b)->rpc_set();
  }
}
```
to the loop that I mentioned:
https://github.com/gnuradio/gnuradio/blob/master/gnuradio-runtime/lib/hier_block2_detail.cc#L800

**Comment by mrjacobagilbert** [comment]
@bastibl - I had tried this (both at L800 and L886) but it segfaults the first time any call is made on the block sptr. Thus my ugly fix.

**Comment by bastibl** [comment]
strange... I tested the snipped that I posted above with your test block (and more complex examples) against maint and next, and didn't experience a segfault.
Anyhow, your code also works.

**Comment by mrjacobagilbert** [comment]
Ok let me try again and make sure I didn't have the OOT built against a different GR version or something.

**Comment by mrjacobagilbert** [comment]
Whoops... should I open a new PR or can this be saved? I verified what @bastibl proposed work and I think that is a much better solution.
